packages/controller/lib/setup.js

```diff
@@ -2725,37 +2729,39 @@ async function delStates() {
     return keys ? keys.length : 0;
 }
 
-function cleanDatabase(isDeleteDb, callback) {
+/**
+ * Cleans the database
+ *
+ * @param {boolean} isDeleteDb - if whole db should be destroyed
+ * @return {Promise<number>}
+ */
+async function cleanDatabase(isDeleteDb) {
     if (isDeleteDb) {
-        objects.destroyDB(async () => {
-            // Clean up states
-            const keysCount = await delStates();
-
-            if (callback) {
-                return void callback(keysCount);
-            }
-        });
+        await objects.destroyDBAsync();
+        // Clean up states
+        return delStates();
     } else {
         // Clean only objects, not the views
-        objects.getObjectList({ startkey: '\u0000', endkey: '\u9999' }, async (err, res) => {
-            let ids = [];
-            if (!err && res.rows.length) {
+        let ids = [];
+
+        try {
+            const res = await objects.getObjectListAsync({ startkey: '\u0000', endkey: '\u9999' });
+            if (res.rows.length) {
                 console.log(`clean ${res.rows.length} objects...`);
                 ids = res.rows.map(e => e.id);
             }
-            await delObjects(ids);
-            // Clean up states
-            const keysCount = await delStates();
+        } catch {
```

Comment:
Does node 12 support omitting the error param yet? Not sure.

Reply:
yes, we do this a lot